bossdata/meta.py

```diff
@@ -98,36 +100,74 @@ def create_meta_lite(sp_all_path, db_path, verbose=True):
             to be a gzipped ASCII data file.
         db_path(str): Local path where the corresponding sqlite3 database will be written.
     """
+
+    # Hardcode to what I think is (quite) conservative value; smaller == slower.
+    # At 50k see memory peak ~2 GB, incldes holding ~750 MB of source in memory.
```

Comment:
This is my (admittedly very rough) accounting, based on looking at memory used by the whole process via top:

I am assuming the gz file is unzipped and in memory; that is roughly 750 Mb.  SQL/DB data shouldn't been in memory, in general, I don't think; or not very much; that is going to disk.  This more or less matches that the memory used by the process when running bottoms out at ~800+ MB.

The remainder (which bounces around a lot but peaks at about 2 GB when I had it doing chunks of 50k records at a time; it does 20k now) must be "the rest of it" including the parsed data/table itself, and whatever magic is doing the parsing.

This points at some additional changes to try out:  writing the data to disk, then reading it back a chunk at a time.  It is hard to understand what all this memory is being used for in parsing, if that is where the rest of it is going; one thing to try here would be to specify the format rather than let it be inferred, and see if this helps